analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/LocksReleasedAllPaths.ReaderWriterLockSlim.cs

```diff
@@ -259,20 +259,140 @@ public void Throw_Finally(object param)
 
         public void IsReadLockHeld()
         {
-            readerWriterLockSlim.EnterReadLock(); // Noncompliant FP, https://github.com/SonarSource/sonar-dotnet/issues/5416
+            readerWriterLockSlim.EnterReadLock();
+            if (readerWriterLockSlim.IsReadLockHeld)    // Compliant, https://github.com/SonarSource/sonar-dotnet/issues/5416
+            {
+                readerWriterLockSlim.ExitReadLock();
+            }
+        }
+
+        public void IsReadLockHeld_NoLocking()
+        {
+            if (readerWriterLockSlim.IsReadLockHeld)    // Noncompliant
```

Comment:
I like the fact that we also report in cases when we did not acquire the lock, and only released on certain paths.  However, it is not aligned with the original description of the rule which states: "If a lock is acquired and released within a method, then it must be released along all execution paths of that method."
I suggest to update the description of the rule.

Reply:
RSPEC will get updated in #5502

Reply:
RSPEC #910